app/src/main/java/org/openmrs/eip/app/utils/ReceiverMessageUtils.java

```diff
@@ -0,0 +1,91 @@
+package org.openmrs.eip.app.utils;
+
+import java.text.MessageFormat;
+import java.time.LocalDateTime;
+
+import org.apache.camel.ProducerTemplate;
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.eip.app.management.entity.AbstractEntity;
+import org.openmrs.eip.app.management.entity.ReceiverSyncRequest;
+import org.openmrs.eip.app.management.entity.SenderSyncResponseModel;
+import org.openmrs.eip.app.management.entity.SyncMessage;
+import org.openmrs.eip.component.Constants;
+import org.openmrs.eip.component.SyncProfiles;
+import org.openmrs.eip.component.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile(SyncProfiles.RECEIVER)
+public class ReceiverMessageUtils {
```

Comment:
This should be a camel processor, i.e. a class that implements Processor, can you change it?

Reply:
Camel has a built-in way to call processor with the processor tag

Reply:
Now I see you are calling this from SiteMessageConsumer which is not a route which is why you made it a bean, I guess it is fine

Reply:
But I find ReceiverMessageUtils strange for a bean class, to me it seems similar to the sender activemq publisher, how about ReceiverActiveMqMessagePublisher? Or ReceiverActiveMqMessageSender?